VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitNavigationActivity.java

```diff
@@ -106,32 +108,57 @@ public void onPageScrollStateChanged(int state) {
         mPreviousBtn.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                int index = pager.getCurrentItem();
-                if (index > 0) {
-                    PageViewStateCallback curView = (PageViewStateCallback) pagerAdapter.instantiateItem(pager, index);
-                    if (curView != null)
-                        curView.onPageDisappear();
-                    pager.setCurrentItem(index - 1);
-                }
+                navigatePreviousComponent();
             }
         });
         mNextBtn.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                int index = pager.getCurrentItem();
-                if (index < pagerAdapter.getCount() - 1) {
-                    PageViewStateCallback curView = (PageViewStateCallback) pagerAdapter.instantiateItem(pager, index);
-                    if (curView != null)
-                        curView.onPageDisappear();
-                    pager.setCurrentItem(index + 1);
-                }
+                navigateNextComponent();
             }
         });
+    }
 
+    @Override
+    protected void onResume() {
+        super.onResume();
         updateUIForOrientation();
     }
 
     @Override
+    public boolean navigatePreviousComponent() {
+        boolean ret = false;
+
+        int index = pager.getCurrentItem();
+        if (index > 0) {
+            PageViewStateCallback curView = (PageViewStateCallback) pagerAdapter.instantiateItem(pager, index);
+            if (curView != null)
+                curView.onPageDisappear();
+            pager.setCurrentItem(index - 1);
+            ret = true;
+        }
+
```

Comment:
Yes, also I just merged Vfrom master (and pushed), and I lost a tad bit of
my work. Recreating now… sorry about that.

On Tue, Aug 2, 2016 at 3:33 PM, Christopher Lee <notifications@github.com>
wrote:

> In
> VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitNavigationActivity.java
> <https://github.com/edx/edx-app-android/pull/761#discussion_r73221848>:
>
> >          updateUIForOrientation();
> >      }
> >
> >      @Override
> > +    public boolean navigatePreviousComponent() {
> > +        boolean ret = false;
> > +
> > +        int index = pager.getCurrentItem();
> > +        if (index > 0) {
> > +            PageViewStateCallback curView = (PageViewStateCallback) pagerAdapter.instantiateItem(pager, index);
> > +            if (curView != null)
> > +                curView.onPageDisappear();
> > +            pager.setCurrentItem(index - 1);
> > +            ret = true;
> > +        }
> > +
>
> Could you remove all of these empty lines in this method and below
> (navigateNextComponent)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/edx/edx-app-android/pull/761/files/74aa7add97dd18b9508daecea42bb12965de956f#r73221848>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AR10T-i_MEca1ci-ezCaUgxqfcvcyRsXks5qb5ukgaJpZM4JCZsX>
> .
>



-- 
Regards,

Marc DiNino
Android Engineer
EDX